clippy_lints/src/operators/const_comparisons.rs

```diff
@@ -0,0 +1,209 @@
+#![allow(clippy::match_same_arms)]
+
+use std::cmp::Ordering;
+
+use clippy_utils::consts;
+use clippy_utils::consts::{ConstEvalLateContext, Constant};
+use if_chain::if_chain;
+use rustc_hir::{BinOpKind, Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{layout::HasTyCtxt, Ty, TypeckResults};
+use rustc_span::source_map::{Span, Spanned};
+
+use clippy_utils::diagnostics::span_lint_and_note;
+use clippy_utils::source::snippet;
+use clippy_utils::SpanlessEq;
+
+use super::IMPOSSIBLE_COMPARISONS;
+use super::REDUNDANT_COMPARISONS;
+
+// Extract a comparison between a const and non-const
+// Flip yoda conditionals, turnings expressions like `42 < x` into `x > 42`
+fn comparison_to_const<'tcx>(
+    ctx: &mut ConstEvalLateContext<'_, 'tcx>,
```

Comment:
nit: Can we not use `consts::constant` instead? It might be marginally slower but `ConstEvalLateContext::new` [does nothing spectacular](https://doc.rust-lang.org/nightly/nightly-rustc/src/clippy_utils/consts.rs.html#334-342) so it's at most a micro-optimization. It would just be cleaner.